gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowConfigResourceLocalHandler.java

```diff
@@ -243,7 +243,16 @@ public static FlowSpec createFlowSpecForConfig(FlowConfig flowConfig) {
       // If it is not a run-once job, we should not add flow execution id here,
       // because execution id is generated for every scheduled execution of the flow and cannot be materialized to
       // the flow catalog. In this case, this id is added during flow compilation.
-      configBuilder.addPrimitive(ConfigurationKeys.FLOW_EXECUTION_ID_KEY, String.valueOf(System.currentTimeMillis()));
+      String flowExecutionId;
+      if (flowConfig.getProperties().containsKey(ConfigurationKeys.FLOW_EXECUTION_ID_KEY)) {
```

Comment:
Add a comment here that FLOW_EXECUTION_ID may already be present in FlowSpec in cases where the FlowSpec is forwarded by a slave to the master.